### PR TITLE
Remove torch.no_grad in feature lobes

### PR DESCRIPTION
**Pull request by Gastron**
See discussion in [https://speechbrain.discourse.group/t/use-pretrained-speechbrain-model-as-loss/493/8](https://speechbrain.discourse.group/t/use-pretrained-speechbrain-model-as-loss/493/8). Also discussed this offline with Mirco and Titouan.

This change should theoretically be small, but in practice it affects a huge number of recipes, etc. Should maybe do some timing tests as well as run some basic recipe just to make sure there's no regression.

### CODE REVIEW

**Comment by mravanelli** [comment]
Maybe we can take this opportunity to also replace `.no_grad` in `core.py `(https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/core.py#L1062) with `torch.inference_mode()`. What do you think? I'm not sure if this feature was available in pytorch 1.8 though.

**Comment by mravanelli** [comment]
I tested in on TIMIT and it looks like speed and memory is the same. As for the `torch.inference` part, I think this should be eventually changed in another PR. This could be a big change that will break compatibility with pytorch <=1.9.
